bowling/file.cpp

```diff
@@ -0,0 +1,101 @@
+#include "file.hpp"
+
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <iterator>
+#include <map>
+#include <regex>
+#include <sstream>
+
+File::File(std::string fileName)
+    : fileName_(fileName) {}
+
+std::map<std::string, std::vector<std::pair<int, int>>> File::getPlayerResults() const {
+    return playerResults_;
+}
+
+void File::splitString(const std::string& str, std::vector<std::string>& vec, char delimiter) {
+    std::stringstream ss(str);
+    std::string token;
+    while (std::getline(ss, token, delimiter)) {
+        vec.push_back(token);
+    }
+}
+
+void File::divideResultsIntoPlayers(const std::vector<std::string>& line) {
+    std::vector<std::string> temp{};
+    std::vector<std::pair<int, int>> scoreVector;
+    for (auto singlePlayerResults : line) {
+        temp.clear();
+        this->splitString(singlePlayerResults, temp, ':');
+        scoreVector = this->translateResultsToScoreVector(temp[1]);
+        playerResults_[temp[0]] = scoreVector;
+    }
+}
+
+std::vector<std::pair<int, int>> File::translateResultsToScoreVector(const std::string& results) {
+    std::vector<std::string> results_divided;
+    std::vector<int> temp{};
+    std::vector<std::pair<int, int>> score;
+
+    this->splitString(results, results_divided, '|');
+
+    for (auto elem : results_divided) {
+        if (elem.empty()) {
+            continue;
+        } else if (elem == "X") {
+            score.push_back(std::make_pair(10, 0));
+        } else if (elem == "XX") {
+            score.push_back(std::make_pair(10, 0));
+            score.push_back(std::make_pair(10, 0));
+        } else if (elem == "--") {
+            score.push_back(std::make_pair(0, 0));
+        } else {
+            for (int i = 0; i < 2; i++) {
+                if (elem[i] == '-') {
+                    temp.push_back(0);
+                } else if (std::isdigit(elem[i])) {
+                    temp.push_back((int)elem[i] - 48);

```

Comment:
what is this magic? :)
just use this one:
`elem[i] - '0'`
It's already in int.

Reply:
done